src/contexts/VideoContext.js

```diff
@@ -14,7 +15,21 @@ const videoReducer = (state, action) => {
 
 const VideoProvider = ({ children }) => {
   const [videos, dispatchVideos] = useReducer(videoReducer, []);
-  return <VideoContext.Provider value={{ videos, dispatchVideos }}>{children}</VideoContext.Provider>;
+
+  const getVideo = async (id) => {
+    try {
+      const response = await axios.get(`/api/video/${id}`);
```

Comment:
suggestion - since you're maintaining the API services separately, you can do it here as well

Reply:
thanks. created separate service file for this.